### PR TITLE
benefits - include equity weighted benefit calculation

### PR DESCRIPTION
**Pull request by panosatha**
In order to include the equity weighted benefits I needed to do quite some changes, so I decided to tackle issue #301 at the same time, by changing the method of the aggregation areas, and using generalized static methods to not repeat the same code.

### CODE REVIEW

**Comment by panosatha** [comment]
I commited some changes in the test_database in this pull request, which **do not** need to be reviewed since we use the SVN.

File: tests/test_database/charleston/static/site/site_without_river.toml
Comment (LuukBlom) [inline tests/test_database/charleston/static/site/site_without_river.toml:1]: Are these changes on the SVN as well? I think it'd be nice to keep the SVN updated and refer to/copy the test_database from there.

File: tests/test_object_model/test_benefits.py
Comment (LuukBlom) [inline tests/test_object_model/test_benefits.py:1]: I think we should refactor some of the testing in the beginning of next year. This is okay for now, but ideally, unittests should be much smaller and test one aspect of a function. At the moment you are testing the run_benefit_analysis function as a whole, and if anything goes wrong anywhere in the function, it is quite difficult/tedious to find out what exactly. 

Good practice to follow when unit testing: write 1 assert per test, and write at least 1 test per new function defined in this file (likely more than one to test all exceptions and edge cases. Each test is then named 'test_methodName_givenInputs_shouldDoSomething()', which if any test fails, immediately tells you where to look.
When we do the testing refactor, I'll show/explain the unittest template I made to reduce code duplication and keep organized when writing loads of tests) 

So TODO, writing smaller tests for each function like: 
testing 'has_run_check()'
- test_hasRunCheck_unfinishedRun_false()
- test_hasRunCheck_finishedRun_true()
- test_hasRunCheck_finishedRun_htmlCreated()
- test_hasRunCheck_invalidInputfiles_raiseFileNotFoundError()
etc for other functions

**Review by LuukBlom — COMMENTED** [review]
The code in benefits is well structured and I haven't found a single bug! (although I havent thoroughly tested it) 
The comment about testing is meant more as information than as a suggestion to be implementated now, as we'll refactor testing later.

File: flood_adapt/object_model/benefit.py [hunk 1]
```diff
@@ -145,9 +141,105 @@ def run_cost_benefit(self):
...
+        benefits = benefits.interpolate(method="linear")
+
+        # Calculate benefits
+        benefits["benefits"] = (
+            benefits["risk_no_measures"] - benefits["risk_with_strategy"]
+        )
+        # Calculate discounted benefits using the provided discount rate
+        benefits["benefits_discounted"] = benefits["benefits"] / (
+            1 + discount_rate
+        ) ** (benefits.index - benefits.index[0])
```
[the remark this answers is not in the document]
Comment (panosatha) [reply flood_adapt/object_model/benefit.py:197]: I have now added that in the docstring

File: tests/test_database/charleston/static/site/site_without_river.toml
Comment (panosatha) [reply tests/test_database/charleston/static/site/site_without_river.toml:1]: indeed these will be updated in the SVN as well

File: tests/test_object_model/test_benefits.py
Comment (panosatha) [reply tests/test_object_model/test_benefits.py:1]: Really happy to follow that approach when we do the unit test refactoring

**Review by LuukBlom — APPROVED** [review]
